src/io/flutter/jxbrowser/EmbeddedBrowser.java

```diff
@@ -128,6 +139,12 @@ public void openPanel(ContentManager contentManager, String tabName, String url,
   }
 
   public void updatePanelToWidget(String widgetId) {
-    browser.navigation().loadUrl(mainUrl + "&inspectorRef=" + widgetId);
+    AsyncUtils.whenCompleteUiThread(mainUrlFuture, (url, ex) -> {
+      if (ex != null) {
+        LOG.error(ex);
+        return;
+      }
+      browser.navigation().loadUrl(url + "&inspectorRef=" + widgetId);
```

Comment:
you probably want to avoid navigating if the url is null which is what you would probably want to have the future return if the  server was never oppened.